ssl/ssl_lib.c

```diff
@@ -1186,6 +1186,8 @@ void SSL_free(SSL *s)
     OPENSSL_free(s->ext.alpn);
     OPENSSL_free(s->ext.tls13_cookie);
     OPENSSL_free(s->clienthello);
+    OPENSSL_free(s->pha_context);
+    EVP_MD_CTX_free(s->pha_dgst);
```

Comment:
But in the case of `s->pha_dgst`, it might be necessary to free in `SSL_clear()` since the SSL object could be reused for a new handshake. The use of `pha_dgst` assumes that it is allocated exactly once per handshake after the first Finished exchange.
`pha_context` is less critical, as it's reallocated as needed.